### PR TITLE
Updated folder creation/structure to simplify

### PR DESCRIPTION
**Pull request by bushe**
Updated folder creation so if Source and Public (or alternative) exists it will not recreate Private and Tests paths.

Updated the Public path to check if .ps1 files exist in Source root, if Public folder exists it takes priority.

### CODE REVIEW

File: ConvertTo-Module.ps1 [hunk 1]
```diff
@@ -223,7 +236,15 @@ $Module | Out-File $OutModule -Force
 #Create manifest
 $OutManifest = Join-Path -Path $Path -ChildPath "$ModuleName.psd1"
 $ManifestSplat = $PSBoundParameters
-$ManifestSplat.Remove("Name") | Out-Null
+
+Write-Verbose "Removing any Parameters not supported by Update-ModuleManifest"
```
Comment (martin9700) [inline ConvertTo-Module.ps1]: Love this part!

File: ConvertTo-Module.ps1 [hunk 2]
```diff
@@ -75,7 +75,8 @@ Param (
     [string]$HelpInfoUri,
     [version]$ModuleVersion,
     [object[]]$RequiredModules,
-    [version]$PowerShellVersion
+    [version]$PowerShellVersion,
+    [switch]$ForceFolderCreation
```
Comment (martin9700) [inline ConvertTo-Module.ps1]: No help for ForceFolderCreation... does this create the folder structure like the old behavior?

File: ConvertTo-Module.ps1 [hunk 3]
```diff
@@ -93,16 +94,26 @@ Else
...
+    if($SubPath -eq "Public" -and -not (Test-Path $tempPath.Value) -and (Get-ChildItem $SourcePath\*.ps1).count -ge 1){
+        Write-Verbose "No Public folder found so Defaulting to .ps1 stored directly in source"
+        Set-Variable -Name "$SubPath`Path" -Value $SourcePath
+    }
+    
     If (-not (Test-Path $tempPath.Value))
     {
         New-Item -Path $tempPath.Value -ItemType Directory | Out-Null
     }
+    elseif($ParentPath -eq $SourcePath -and $ForceFolderCreation -eq $false)
```
Comment (martin9700) [inline ConvertTo-Module.ps1]: Instead of ElseIf, why not just 
`If (-not $ForceFolderCreation) { Break }`

**Comment by martin9700** [comment]
Looks good, just add the cbh for $ForcedFolderCreation and we're good to go.

**Comment by martin9700** [comment]
Of course, I have no idea how to do that!!  lol

**Comment by bushe** [comment]
Should be all set, I pushed a commit that included the help.

**Pull request merged**